src/plugin/duration/index.js

```diff
@@ -257,6 +257,24 @@ class Duration {
   asYears() { return this.as('years') }
 }
 
+const addDuration = (date, duration) =>
+  date.add(duration.years(), 'year')
+    .add(duration.months(), 'month')
+    .add(duration.days(), 'day')
+    .add(duration.hours(), 'hour')
+    .add(duration.minutes(), 'minute')
+    .add(duration.seconds(), 'second')
+    .add(duration.milliseconds(), 'millisecond')
```

Comment:
a little trick to minimize the code, pass the third argument to combine add and subtract together.
```
// manipulateDuration(this, value, 1)
// manipulateDuration(this, value, -1)

// e.g. .add(duration.months() * (-1), 'month')
```

also, one thing that confused me, is why adding `value.asMilliseconds()`  is not the correct result. 
It should be the real `ms` between the two dates.

Reply:
thanks for the tip to improve the code
value.asMilliseconds() executed without date context and In this case, month is equal to 30 days. I think it's correct. (moment js has same behavior)